pkg/ui/workspaces/cluster-ui/src/api/clusterSettingsApi.ts

```diff
@@ -20,11 +24,99 @@ export function getClusterSettings(
   req: SettingsRequestMessage,
   timeout: string,
 ): Promise<SettingsResponseMessage> {
+  const urlParams = new URLSearchParams();
+  urlParams.append("unredacted_values", "true");
+  if (req.keys) {
+    urlParams.append("keys", req.keys.join(","));
+  }
+
   return fetchData(
     cockroach.server.serverpb.SettingsResponse,
-    `${ADMIN_API_PREFIX}/settings?unredacted_values=true`,
-    cockroach.server.serverpb.SettingsRequest,
-    req,
+    `${ADMIN_API_PREFIX}/settings?` + urlParams.toString(),
+    null,
+    null,
     timeout,
   );
 }
+
+// Preferred.
+export type GetClusterSettingRequest = {
+  names: string[];
+};
+
+enum ClusterSettingType {
+  BOOLEAN = "b",
+  DURATION = "d",
+  BYTE_SIZE = "z",
+  INTEGER = "i",
+  ENUM = "e",
+  FLOAT = "f,",
+  STRING = "s",
+  VERSION = "m",
+  UNKNOWN = "",
+}
+
+export type ClusterSetting = {
+  name: string;
+  value: string;
+  type: ClusterSettingType;
+  description: string;
+  lastUpdated: moment.Moment | null;
+};
+
+const formatProtoResponse = (
+  data: SettingsResponseMessage,
+): Record<string, ClusterSetting> => {
+  const settingsMap = {} as Record<string, ClusterSetting>;
+
+  const entries = Object.values(data?.key_values);
+  if (entries?.length === 0) {
+    return settingsMap;
+  }
+
+  entries.forEach(kv => {
+    settingsMap[kv.name] = {
+      name: kv.name,
+      value: kv.value,
+      type: kv.type as ClusterSettingType,
+      description: kv.description,
+      lastUpdated: TimestampToMoment(kv.last_updated),
+    };
+  });
+
+  return settingsMap;
+};
+
+const emptyClusterSetting: ClusterSetting = {
+  name: "",
+  value: "",
+  type: ClusterSettingType.UNKNOWN,
+  description: "",
+  lastUpdated: null,
+};
+
+export const useClusterSettings = (req: GetClusterSettingRequest) => {
+  const protoReq = new cockroach.server.serverpb.SettingsRequest({
+    keys: req.names,
+  });
+  const { data, isLoading, error } = useSWRImmutable(req.names, () =>
+    getClusterSettings(protoReq, "1M").then(formatProtoResponse),
```

Comment:
is `"1M"` translated into a duration object somewhere? i think in durations, `M` is a month, where as `m` is minute. Im assuming we want the later?